front/src/applications/editor/tools/switchEdition/components/TrackSectionEndpointSelector.tsx

```diff
@@ -29,6 +29,18 @@ const TrackSectionEndpointSelector = ({ schema, formData, onChange, name }: Fiel
   const { state, setState } = useContext(
     EditorContext
   ) as ExtendedEditorContextType<SwitchEditionState>;
+
+  const duplicateWith = useMemo(() => {
+    const ports = Object.entries(state.entity.properties?.ports ?? {}).map(([k, v]) => ({
+      ...v,
+      port: k,
+      name: `port::${k}`,
+    }));
+    const currentPort = ports.find((p) => p.name === name);
+    const othersPorts = ports.filter((p) => p.name !== name);
+    return othersPorts.filter((p) => p.track === currentPort?.track);
+  }, [state.entity.properties]);
```

Comment:
I think we shouldn't have the warning when entering the form and there isn't any track selected for the ports. 

![Capture d’écran 2024-01-18 à 15 26 40](https://github.com/osrd-project/osrd/assets/113104137/61378284-41cb-4cf3-b2d5-68bbacdfeee1)

You could add : 

```suggestion
  const duplicateWith = useMemo(() => {
    const ports = Object.entries(state.entity.properties?.ports ?? {}).map(([k, v]) => ({
      ...v,
      port: k,
      name: `port::${k}`,
    }));
    if (ports.every((port) => !port.track)) return [];
    const currentPort = ports.find((p) => p.name === name);
    const othersPorts = ports.filter((p) => p.name !== name);
    return othersPorts.filter((p) => p.track === currentPort?.track);
  }, [state.entity.properties]);
```

Reply:
Would like your opinion on that too @bloussou :)

Reply:
Nice spot. Indeed we only want the warning when ports are filled.